tsinfer/formats.py

```diff
@@ -226,15 +226,24 @@ def zarr_summary(array):
     return ret
 
 
-def chunk_iterator(array):
+def chunk_iterator(array, indexes=None):
     """
-    Utility to iterate over the rows in the specified array efficiently
-    by accessing one chunk at a time.
+    Utility to iterate over closely spaced rows in the specified array efficiently
+    by accessing one chunk at a time (normally used as an iterator over each row)
     """
+    if indexes is None:
+        indexes = range(array.shape[0])
+    else:
+        if len(indexes) > 0 and (indexes[0] < 0 or np.any(np.diff(indexes) <= 0)):
+            raise ValueError("ids must be positive and in ascending order")
```

Comment:
Since we are ensuring the indices are mononically increasing, think we can simply check the top and bottom ends, so I'll add a check to make sure that `indexes[-1] >= array.shape[0]` too.

Reply:
Done